modules/effects/arrow-effects/src/main/kotlin/arrow/effects/IORunLoop.kt

```diff
@@ -133,31 +133,27 @@ internal object IORunLoop {
   private fun <A> suspendInAsync(
     currentIO: IO<A>,
     bFirst: BindF?,
-    bRest: CallStack?,
-    register: (IOConnection, (Either<Throwable, Any?>) -> Unit) -> Unit): IO<A> =
+    bRest: CallStack?): IO<A> =
   // Hitting an async boundary means we have to stop, however
   // if we had previous `flatMap` operations then we need to resume
   // the loop with the collected stack
     when {
       bFirst != null || (bRest != null && bRest.isNotEmpty()) ->
         IO.Async { conn, cb ->
-          val rcb = RestartCallback(cb as Callback, null)
-          rcb.prepare(bFirst, bRest)
-          register(conn, rcb)
+          loop(currentIO, conn, cb as Callback, null, bFirst, bRest)
```

Comment:
why was this so different than cats? I'm not 100% sure of this change, but this is what you can find in cats for this function